subprojects/core/src/integTest/groovy/org/gradle/api/tasks/TaskInputPropertiesIntegrationTest.groovy

```diff
@@ -417,31 +414,129 @@ apply from:'scriptPlugin.gradle'
             class MyTask extends DefaultTask {
                 @Input
                 FooType foo
+            }
+
+            task createFile(type: MyTask) {
+                ext.outputFile = file('output.txt')
+                outputs.file(outputFile)
+
+                foo = FooType.FOO
+
+                doLast {
```

Comment:
Should be a `@TaskAction` in `MyTask` to have all the imperative code in the task definition.